### PR TITLE
Replacing teapot's configuration reconfigures the session.

### PR DESCRIPTION
**Pull request by elland**
Ideally even changing some configuration properties should trigger this. Ideas?

### CODE REVIEW

**Comment by designatednerd** [comment]
What is the issue you're trying to solve here - is changing properties on the configuration not actually updating the session itself? 

I also don't think there's a great way to have changing properties make this change without wrapping whatever properties of the session configuration you want to change in convenience vars, unfortunately

**Comment by elland** [comment]
> What is the issue you're trying to solve here 

The particular use-case I have at the moment is injecting a URL protocol class into the configuration to hijack some of the requests. 

> is changing properties on the configuration not actually updating the session itself?

No. They don't.

> I also don't think there's a great way to have changing properties make this change without wrapping whatever properties of the session configuration you want to change in convenience vars, unfortunately

Not the prettiest solution, but I think you're right. Since `URLSessionConfiguration` is an object we won't get any callbacks when inner values change without KVO.

**Comment by elland** [comment]
Formatting is hard. :D

File: Teapot/Teapot.swift [hunk 1]
```diff
@@ -274,7 +276,7 @@ open class Teapot {
 
         if let headerFields = headerFields {
             for headerField in headerFields {
-                if headerField.key.lowercased() == "content-type" {
+                if headerField.key == "Content-Type" {
```
Comment (designatednerd) [inline Teapot/Teapot.swift]: I would strongly recommend leaving this - that way if someone does `Content-type` we still catch the fact that it has a content type

File: Teapot/Teapot.swift [hunk 2]
```diff
@@ -239,12 +241,12 @@ open class Teapot {
...
-                || TEAPOT - REQUEST CONSTRUCTION ERROR
-                || Could not get components for path \"\(String(describing: path))\"
-                ||
+                    ||
+                    || TEAPOT - REQUEST CONSTRUCTION ERROR
+                    || Could not get components for path \"\(String(describing: path))\"
+                    ||
 
-                """)
+                    """)
```
Comment (designatednerd) [inline Teapot/Teapot.swift]: why did these need to be re-indented?

File: Teapot/Teapot.swift [hunk 3]
```diff
@@ -18,21 +18,15 @@ open class Teapot {
         DispatchQueue(label: "com.backkenbaeck.Teapot.RunTaskQueue", qos: .background)
     }()
 
-    open lazy var queue: OperationQueue = {
-        let queue = OperationQueue()
-        queue.name = "no.bakkenbaeck.NetworkQueue"
-        queue.qualityOfService = .userInitiated
+    open var queue: OperationQueue
```
Comment (designatednerd) [inline Teapot/Teapot.swift:21]: why does this need to be a `var` instead of just a `let`?

File: Teapot/Teapot.swift (hunk 1 above)
Comment (elland) [reply Teapot/Teapot.swift]: I think I accidentally reverted this when merging back from master. 🙈

File: Teapot/Teapot.swift (hunk 3 above)
Comment (elland) [reply Teapot/Teapot.swift:21]: Because otherwise it can't be `open`. `let`s are implicitly `final`.

**Pull request closed without merging**